content/ytt/docs/latest/yaml-primer.md

```diff
@@ -0,0 +1,195 @@
+
+With this structure in mind, we can now look into how `ytt` annotates it.
+
+## Annotating YAML
+
+A `ytt` "annotation" is a YAML comment of a specific format; it attaches behavior to a node.
+
+For example:
+
+![](/images/ytt/yaml-primer/map-item-with-value.jpg)
+<!-- source: https://miro.com/app/board/o9J_lIfcKZY=/?moveToWidget=3074457358161105960&cot=14 -->
+_Figure 9: A YAML file with a `ytt` annotation._
+
+Can be understood to mean that the value of `foo` is the result of evaluating the expression `13 + 23 + 6`.
+
+The diagram reveals how the annotation is attached to the YAML tree:
+
+![](/images/ytt/yaml-primer/map-item-ann-with-value-ast.jpg)
+<!-- source: https://miro.com/app/board/o9J_lIfcKZY=/?moveToWidget=3074457358161105960&cot=14 -->
+_Figure 10: The annotation (dotted black) is attached to the **map item** (solid green)._
+
+This is a document, whose value is a map, that contains a single map item. To the right of the map item is an annotation that contains an expression that, when evaluated, becomes a _value_. The annotation is attached to the map item.
+
+### How an Annotation Finds its Node
+
+When attaching annotations, `ytt` follows these two rules (in order):
+1. the annotation is attached to the value-holding node **on its left**, if there is one; otherwise,
+2. the annotation is attached to the value-holding node **directly below** it.
+
+In the example above, the value-holding node to the left of the annotation is the map item.
+
+As noted in the summary [above](#summary), "value-holding" nodes are:
+- **Document**s,
+- **Map Item**s, and
+- **Array Item**s.
+
+Let's see these rules at play:
+
+![](/images/ytt/yaml-primer/overlay-ann-on-doc-and-map.jpg)
+<!-- source: https://miro.com/app/board/o9J_lIfcKZY=/?moveToWidget=3074457358161105960&cot=14 -->
+_Figure 11: A templated document with three annotations._
+
+We'll ignore _what_ these annotations mean, for now, and focus on _where_ they attach. There are three annotations in total.
+
+Visualizing the YAML tree can help:
+
+
+![](/images/ytt/yaml-primer/overlay-ann-on-doc-and-map-ast.jpg)
+<!-- source: https://miro.com/app/board/o9J_lIfcKZY=/?moveToWidget=3074457358161105960&cot=14 -->
+_Figure 12: Each annotation attaches to the value-holding node to its left or bottom._
+
+Taking each annotation in turn:
+- `@overlay/match by=overlay.all`:
+  - is in the document set, but that is not a value-holding node
+  - has a **document** node just below it and so is attached to _that_ node.
+- `@overlay/replace`:
+  - has no node to its left;
+  - has a **map** just below it but maps are _not_ value-holding;
+  - the next node is a **map item**, and so attaches to _that_ node.
+- `@ 13 + 23 + 6`:
+  - has a **map item** to its left, and so attaches to _that_ node.
+  - there _is_ another **map item** below it (i.e. `bar: true`), but a home has already been found for this annotation.
```

Comment:
is it useful to showcase array example as well? some folks get hung up on `- #@ ...`

Reply:
done.